test/functional/outputs/forward_to_cloudwatch_test.go

```diff
@@ -0,0 +1,191 @@
+package outputs
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/ViaQ/logerr/log"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	cwl "github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+
+	openshiftv1 "github.com/openshift/api/route/v1"
+
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/runtime"
+)
+
+var _ = Describe("[Functional][Outputs][CloudWatch] FluentdForward Output to CloudWatch", func() {
+
+	const (
+		logSize   = 128
+		numOfLogs = 8
+
+		// examples from AWS docs
+		awsAccessKeyID     = "AKIAIOSFODNN7EXAMPLE"
+		awsSecretAccessKey = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY" //nolint:gosec
+	)
+
+	var (
+		framework *functional.FluentdFunctionalFramework
+
+		addMotoContainerVisitor = func(b *runtime.PodBuilder) error {
+			log.V(2).Info("Adding AWS CloudWatch Logs mock container")
+			b.AddContainer("moto", "quay.io/openshift-logging/moto:2.2.3.dev0").
+				WithCmdArgs([]string{"-s"}).
+				End()
+			return nil
+		}
+
+		mountCloudwatchSecretVisitor = func(b *runtime.PodBuilder) error {
+			log.V(2).Info("Mounting cloudwatch secret to the fluentd container")
+			b.AddSecretVolume("cloudwatch", "cloudwatch").
+				GetContainer(constants.FluentdName).
+				AddVolumeMount("cloudwatch", "/var/run/ocp-collector/secrets/cloudwatch", "", true)
+			return nil
+		}
+
+		cwlClient *cwl.Client
+
+		maxDuration, _          = time.ParseDuration("10m")
+		defaultRetryInterval, _ = time.ParseDuration("10s")
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewFluentdFunctionalFramework()
+
+		log.V(2).Info("Creating service moto")
+		service := runtime.NewService(framework.Namespace, "moto")
```

Comment:
This looks like a pattern we are repeating from the Loki tests and were started as part of the kafka functional tests.  Functional framework intent is to create a single POD that includes all needed containers for the test.  This means they should share the same network address space.  

* Why do we need a service and route to create a test that writes from the forwarder to cloudwatch mock.   
* What are we missing in the framework that keeps us from using it as designed.  
* What do we gain be introducing the additional complexities of OpenShift

Reply:
It's not the writing to the mock, it's the reading from the mock. The test, running outside of cluster, talks to moto to check that the logs made it there. That's what the `Context("When sending a sequence of log messages to CloudWatch"` clause does.
To fit this into the framework we could
- fork moto and trying to merge it upstream to get it to log what we need into its own log which we can `oc rsh cat`
- put the aws cli client or write our own simple aws cli client that dumps into its log what it reads from moto, put it into a container, deploy into the test pod. Dealing with that container image looked like a pain to me.